tcms/core/tests/test_ajax.py

```diff
@@ -185,3 +182,48 @@ def test_remove_tag_from_test_run(self):
 
         self.assertEqual(response.status_code, http.client.OK)
         self.assertEqual(self.test_run.tag.count(), 0)
+
+
+class Test_Tag_Render(Test_Tag_Test):
+
+    @classmethod
+    def setUpTestData(cls):
+        super(Test_Tag_Render, cls).setUpTestData()
+
+        cls.test_plan.add_tag(cls.test_tag)
+        cls.test_case.add_tag(cls.test_tag)
+        cls.test_run.add_tag(cls.test_tag)
+
+        for _ in range(0, 3):
+            TestPlanFactory().add_tag(cls.test_tag)
+
+        for _ in range(0, 4):
+            TestCaseFactory().add_tag(cls.test_tag)
+
+        for _ in range(0, 5):
+            TestRunFactory().add_tag(cls.test_tag)
+
+    def test_render_plan(self):
+        response = self.client.get(self.url, {
+            'plan': self.test_plan.plan_id
+        })
+
+        self._assert_tags(response)
+
+    def test_render_case(self):
+        response = self.client.get(self.url, {
+            'case': self.test_case.case_id
+        })
+
+        self._assert_tags(response)
+
+    def _assert_tags(self, response):
+        self.assertEqual(response.status_code, http.client.OK)
+
+        self.assertEqual(self.test_plan.tag.count(), 1)
+        self.assertEqual(self.test_case.tag.count(), 1)
+        self.assertEqual(self.test_run.tag.count(), 1)
+
```

Comment:
Between the above 3 lines and the below 3 lines we need some kind of comment to tell us what's going on. At first I though the above 3 lines were pointless. Then I saw that you already have the tags attached to the primary objects (case/plan/run) and then attach to some more which leads to the +1 change in the numbers below. 

This isn't quite obvious initially and warrants a comment. 

Everything else looks good and is ready for merge.